--- kdebase-runtime-4.0.10~svn783501.orig/doc/userguide/kde-for-admins.docbook
+++ kdebase-runtime-4.0.10~svn783501/doc/userguide/kde-for-admins.docbook
@@ -1765,8 +1765,8 @@
 whose icon and name are defined by <filename>kde-settings.directory</filename></para>
 </listitem>
 <listitem><para>Info Center has a hidden Information menu whose
-contents are defined by <filename>kde-information.menu</filename> and
-whose icon and  name are  defined by <filename>kde-information.directory</filename>.</para>
+contents are defined by <filename>kde4-information.menu</filename> and
+whose icon and  name are  defined by <filename>kde4-information.directory</filename>.</para>
 </listitem>
 
 <listitem><para>Screensavers contains a hidden System/Screensavers menu,
--- kdebase-runtime-4.0.10~svn783501.orig/kcontrol/menus/CMakeLists.txt
+++ kdebase-runtime-4.0.10~svn783501/kcontrol/menus/CMakeLists.txt
@@ -4,5 +4,5 @@
 ########### install files ###############
 
 install( FILES kde-information.menu
-    DESTINATION ${SYSCONF_INSTALL_DIR}/xdg/menus/)
-install( FILES kde-information.directory  DESTINATION  ${XDG_DIRECTORY_INSTALL_DIR} )
+    DESTINATION ${SYSCONF_INSTALL_DIR}/xdg/menus/ RENAME kde4-information.menu )
+install( FILES kde-information.directory  DESTINATION  ${XDG_DIRECTORY_INSTALL_DIR} RENAME kde4-information.directory )
--- kdebase-runtime-4.0.10~svn783501.orig/kcontrol/menus/kde-information.menu
+++ kdebase-runtime-4.0.10~svn783501/kcontrol/menus/kde-information.menu
@@ -3,7 +3,7 @@
 
 <Menu>
 	<Name>Information</Name>
-	<Directory>kde-information.directory</Directory>
+	<Directory>kde4-information.directory</Directory>
 	<Include>
 		<Category>X-KDE-information</Category>
 	</Include>
