Index: runtime/phonon/xine/videowidget.cpp
===================================================================
--- runtime/phonon/xine/videowidget.cpp	(revision 839926)
+++ runtime/phonon/xine/videowidget.cpp	(revision 839927)
@@ -226,12 +226,13 @@
     return m_brightness;
 }
 
+static const qreal ONE = 1.0;
 void VideoWidget::setBrightness(qreal newBrightness)
 {
-    newBrightness = qBound(-1.0, newBrightness, 1.0);
+    newBrightness = qBound(-ONE, newBrightness, ONE);
     if (m_brightness != newBrightness) {
         m_brightness = newBrightness;
-        upstreamEvent(new SetParamEvent(XINE_PARAM_VO_BRIGHTNESS, static_cast<int>(0x7fff * (m_brightness + 1.0))));
+        upstreamEvent(new SetParamEvent(XINE_PARAM_VO_BRIGHTNESS, static_cast<int>(0x7fff * (m_brightness + ONE))));
     }
 }
 
@@ -242,10 +243,10 @@
 
 void VideoWidget::setContrast(qreal newContrast)
 {
-    newContrast = qBound(-1.0, newContrast, 1.0);
+    newContrast = qBound(-ONE, newContrast, ONE);
     if (m_contrast != newContrast) {
         m_contrast = newContrast;
-        upstreamEvent(new SetParamEvent(XINE_PARAM_VO_CONTRAST, static_cast<int>(0x7fff * (m_contrast + 1.0))));
+        upstreamEvent(new SetParamEvent(XINE_PARAM_VO_CONTRAST, static_cast<int>(0x7fff * (m_contrast + ONE))));
     }
 }
 
@@ -256,10 +257,10 @@
 
 void VideoWidget::setHue(qreal newHue)
 {
-    newHue = qBound(-1.0, newHue, 1.0);
+    newHue = qBound(-ONE, newHue, ONE);
     if (m_hue != newHue) {
         m_hue = newHue;
-        upstreamEvent(new SetParamEvent(XINE_PARAM_VO_HUE, static_cast<int>(0x7fff * (m_hue + 1.0))));
+        upstreamEvent(new SetParamEvent(XINE_PARAM_VO_HUE, static_cast<int>(0x7fff * (m_hue + ONE))));
     }
 }
 
@@ -270,10 +271,10 @@
 
 void VideoWidget::setSaturation(qreal newSaturation)
 {
-    newSaturation = qBound(-1.0, newSaturation, 1.0);
+    newSaturation = qBound(-ONE, newSaturation, ONE);
     if (m_saturation != newSaturation) {
         m_saturation = newSaturation;
-        upstreamEvent(new SetParamEvent(XINE_PARAM_VO_SATURATION, static_cast<int>(0x7fff * (m_saturation + 1.0))));
+        upstreamEvent(new SetParamEvent(XINE_PARAM_VO_SATURATION, static_cast<int>(0x7fff * (m_saturation + ONE))));
     }
 }
 
