--- a/khelpcenter/searchhandlers/khc_docbookdig.pl.cmake
+++ b/khelpcenter/searchhandlers/khc_docbookdig.pl.cmake
@@ -26,7 +26,7 @@
 
 use Getopt::Long;
 
-my $htdigdata = "/srv/www/htdig/common/";
+my $htdigdata = "/etc/htdig";
 my $htdigbin = "/usr/bin";
 my $kdeprefix = "@CMAKE_INSTALL_PREFIX@";
 chomp $kdeprefix;
@@ -175,7 +175,7 @@
 max_hop_count:          4
 robotstxt_name:         kdedig
 compression_level:      6
-template_map:           Long long $kdeprefix/share/apps/khelpcenter/searchhandlers/htdig/htdig_long.html
+template_map:           Long long @DATA_INSTALL_DIR@/khelpcenter/searchhandlers/htdig/htdig_long.html
 search_algorithm:       exact:1 prefix:0.8
 maximum_pages:          1
 matches_per_page:       10
--- a/khelpcenter/searchhandlers/khc_htdig.pl.cmake
+++ b/khelpcenter/searchhandlers/khc_htdig.pl.cmake
@@ -26,7 +26,7 @@
 
 use Getopt::Long;
 
-my $htdigdata = "/srv/www/htdig/common/";
+my $htdigdata = "/etc/htdig";
 my $htdigbin = "/usr/bin";
 my $kdeprefix = "@CMAKE_INSTALL_PREFIX@";
 chomp $kdeprefix;
--- a/khelpcenter/searchhandlers/khc_htsearch.pl
+++ b/khelpcenter/searchhandlers/khc_htsearch.pl
@@ -8,7 +8,7 @@
 use open IO => ':utf8';
 use open ':std';
 
-my $htsearchpath="/srv/www/cgi-bin/htsearch";
+my $htsearchpath="/usr/lib/cgi-bin/htsearch";
 
 my $config;
 my $format;
