--- a/kcmshell/CMakeLists.txt
+++ b/kcmshell/CMakeLists.txt
@@ -3,7 +3,7 @@
 
 kde4_add_kdeinit_executable( kcmshell4 ${kcmshell_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kcmshell4 ${KDE4_KUTILS_LIBS} )
+target_link_libraries(kdeinit_kcmshell4 ${KDE4_KUTILS_LIBS} ${QT_QTDBUS_LIBRARY} )
 
 install(TARGETS kdeinit_kcmshell4  ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(TARGETS kcmshell4          ${INSTALL_TARGETS_DEFAULT_ARGS})
--- a/kcontrol/componentchooser/CMakeLists.txt
+++ b/kcontrol/componentchooser/CMakeLists.txt
@@ -21,7 +21,7 @@
 kde4_add_plugin(kcm_componentchooser ${kcm_componentchooser_PART_SRCS})
 
 
-target_link_libraries(kcm_componentchooser  ${KDE4_KIO_LIBS})
+target_link_libraries(kcm_componentchooser  ${KDE4_KIO_LIBS} ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS kcm_componentchooser  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kcontrol/dnssd/CMakeLists.txt
+++ b/kcontrol/dnssd/CMakeLists.txt
@@ -11,7 +11,7 @@
 kde4_add_plugin(kcm_kdnssd ${kcm_kdnssd_PART_SRCS})
 
 
-target_link_libraries(kcm_kdnssd  ${KDE4_KDNSSD_LIBS} )
+target_link_libraries(kcm_kdnssd  ${KDE4_KDNSSD_LIBS} ${KDE4_KDEUI_LIBS} ${QT_QTDBUS_LIBRARY} )
 
 install(TARGETS kcm_kdnssd  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kcontrol/emoticons/CMakeLists.txt
+++ b/kcontrol/emoticons/CMakeLists.txt
@@ -2,7 +2,7 @@
 kde4_add_ui_files(kcmemoticons_SRCS emoticonslist.ui)
 
 kde4_add_plugin(kcm_emoticons ${kcmemoticons_SRCS})
-target_link_libraries(kcm_emoticons ${KDE4_KDEUI_LIBS} ${KDE4_KPARTS_LIBS} ${KDE4_KNEWSTUFF2_LIBS} ${KDE4_KUTILS_LIBRARY})
+target_link_libraries(kcm_emoticons ${KDE4_KDEUI_LIBS} ${KDE4_KPARTS_LIBS} ${KDE4_KNEWSTUFF2_LIBS} ${KDE4_KUTILS_LIBRARY} ${QT_QTXML_LIBRARY})
 
 install(TARGETS kcm_emoticons  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kcontrol/icons/CMakeLists.txt
+++ b/kcontrol/icons/CMakeLists.txt
@@ -9,7 +9,7 @@
 kde4_add_plugin(kcm_icons ${kcm_icons_PART_SRCS})
 
 
-target_link_libraries(kcm_icons ${KDE4_KIO_LIBS} ${KDE4_KNEWSTUFF2_LIBS})
+target_link_libraries(kcm_icons ${KDE4_KIO_LIBS} ${KDE4_KNEWSTUFF2_LIBS} ${QT_QTSVG_LIBRARY})
 
 install(TARGETS kcm_icons  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kcontrol/kded/CMakeLists.txt
+++ b/kcontrol/kded/CMakeLists.txt
@@ -9,7 +9,7 @@
 kde4_add_plugin(kcm_kded ${kcm_kded_PART_SRCS})
 
 
-target_link_libraries(kcm_kded ${KDE4_KDEUI_LIBS})
+target_link_libraries(kcm_kded ${KDE4_KDEUI_LIBS} ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS kcm_kded  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kcontrol/knotify/CMakeLists.txt
+++ b/kcontrol/knotify/CMakeLists.txt
@@ -9,7 +9,7 @@
 kde4_add_plugin(kcm_knotify ${kcm_knotify_PART_SRCS})
 
 
-target_link_libraries(kcm_knotify  ${KDE4_KIO_LIBS} ${KDE4_KNOTIFYCONFIG_LIBS} ${QT_QTGUI_LIBRARY} )
+target_link_libraries(kcm_knotify  ${KDE4_KIO_LIBS} ${KDE4_KNOTIFYCONFIG_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QTDBUS_LIBRARY} )
 
 install(TARGETS kcm_knotify  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/khelpcenter/CMakeLists.txt
+++ b/khelpcenter/CMakeLists.txt
@@ -58,7 +58,7 @@
     install(TARGETS khelpcenter ${INSTALL_TARGETS_DEFAULT_ARGS})
 else (WIN32)
     kde4_add_kdeinit_executable( khelpcenter ${khelpcenter_KDEINIT_SRCS})
-    target_link_libraries(kdeinit_khelpcenter  ${KDE4_KDE3SUPPORT_LIBS} ${KDE4_KHTML_LIBS}  ${QT_QTXML_LIBRARY}  )
+    target_link_libraries(kdeinit_khelpcenter  ${KDE4_KDE3SUPPORT_LIBS} ${KDE4_KHTML_LIBS} ${QT_QTXML_LIBRARY} ${QT_QTDBUS_LIBRARY} )
     install(TARGETS kdeinit_khelpcenter  ${INSTALL_TARGETS_DEFAULT_ARGS} )
     install(TARGETS khelpcenter ${INSTALL_TARGETS_DEFAULT_ARGS})
 endif (WIN32)
--- a/kioslave/desktop/CMakeLists.txt
+++ b/kioslave/desktop/CMakeLists.txt
@@ -4,7 +4,7 @@
 
 kde4_add_plugin(kio_desktop  ${kio_desktop_PART_SRCS})
 
-target_link_libraries(kio_desktop  ${KDE4_KIO_LIBS})
+target_link_libraries(kio_desktop  ${KDE4_KIO_LIBS} ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS kio_desktop DESTINATION  ${PLUGIN_INSTALL_DIR} )
 
--- a/kioslave/remote/kdedmodule/CMakeLists.txt
+++ b/kioslave/remote/kdedmodule/CMakeLists.txt
@@ -9,7 +9,7 @@
 kde4_add_plugin(kded_remotedirnotify ${kded_remotedirnotify_PART_SRCS})
 
 
-target_link_libraries(kded_remotedirnotify  ${KDE4_KIO_LIBS})
+target_link_libraries(kded_remotedirnotify  ${KDE4_KIO_LIBS} ${QT_QTDBUS_LIBRARY} )
 
 install(TARGETS kded_remotedirnotify  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kioslave/thumbnail/CMakeLists.txt
+++ b/kioslave/thumbnail/CMakeLists.txt
@@ -28,7 +28,7 @@
 
 kde4_add_plugin(svgthumbnail ${svgthumbnail_PART_SRCS})
 
-target_link_libraries(svgthumbnail ${KDE4_KDEUI_LIBS}) # needs kdeui for ksvgrenderer
+target_link_libraries(svgthumbnail ${KDE4_KDEUI_LIBS} ${QT_QTSVG_LIBRARY}) # needs kdeui for ksvgrenderer
 
 install(TARGETS svgthumbnail DESTINATION ${PLUGIN_INSTALL_DIR})
 
--- a/kioslave/trash/CMakeLists.txt
+++ b/kioslave/trash/CMakeLists.txt
@@ -14,7 +14,7 @@
 kde4_add_plugin(kio_trash ${kio_trash_PART_SRCS})
 
 
-target_link_libraries(kio_trash  ${KDE4_KIO_LIBS} ${KDE4_SOLID_LIBS})
+target_link_libraries(kio_trash  ${KDE4_KIO_LIBS} ${KDE4_SOLID_LIBS} )
 
 install(TARGETS kio_trash  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kpasswdserver/CMakeLists.txt
+++ b/kpasswdserver/CMakeLists.txt
@@ -2,7 +2,10 @@
 
 kde4_add_plugin(kded_kpasswdserver ${kded_kpasswdserver_SRCS})
 
-target_link_libraries(kded_kpasswdserver  ${KDE4_KIO_LIBS} ${QT_QT3SUPPORT_LIBRARY})
+target_link_libraries(kded_kpasswdserver  ${KDE4_KIO_LIBS} ${QT_QT3SUPPORT_LIBRARY} ${QT_QTDBUS_LIBRARY})
+if (X11_FOUND)
+    target_link_libraries(kded_kpasswdserver ${X11_X11_LIB})
+endif (X11_FOUND)
 
 install(TARGETS kded_kpasswdserver  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kstyles/oxygen/CMakeLists.txt
+++ b/kstyles/oxygen/CMakeLists.txt
@@ -17,7 +17,7 @@
 
 kde4_add_plugin(oxygen ${oxygen_PART_SRCS})
 
-target_link_libraries(oxygen  ${KDE4_KDEUI_LIBS} )
+target_link_libraries(oxygen  ${KDE4_KDEUI_LIBS} ${QT_QTDBUS_LIBRARY} )
 
 install(TARGETS oxygen  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/styles/ )
 
--- a/ktimezoned/CMakeLists.txt
+++ b/ktimezoned/CMakeLists.txt
@@ -5,7 +5,7 @@
 
 #qt4_add_dbus_adaptor(kded_ktimezoned_SRCS org.kde.KTimeZoned ktimezonedbus.h DBusHandler)
 
-target_link_libraries(kded_ktimezoned ${KDE4_KIO_LIBS})
+target_link_libraries(kded_ktimezoned ${KDE4_KIO_LIBS} ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS kded_ktimezoned DESTINATION ${PLUGIN_INSTALL_DIR})
 
--- a/kuiserver/CMakeLists.txt
+++ b/kuiserver/CMakeLists.txt
@@ -20,7 +20,7 @@
 
 kde4_add_kdeinit_executable( kuiserver ${kuiserver_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kuiserver  ${KDE4_KIO_LIBS} )
+target_link_libraries(kdeinit_kuiserver  ${KDE4_KIO_LIBS} ${QT_QTDBUS_LIBRARY} )
 install(TARGETS kdeinit_kuiserver ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 target_link_libraries( kuiserver kdeinit_kuiserver )
--- a/kurifilter-plugins/ikws/CMakeLists.txt
+++ b/kurifilter-plugins/ikws/CMakeLists.txt
@@ -6,7 +6,7 @@
 
 kde4_add_plugin(kuriikwsfilter ${kuriikwsfilter_PART_SRCS})
 
-target_link_libraries(kuriikwsfilter ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS})
+target_link_libraries(kuriikwsfilter ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS} ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS kuriikwsfilter DESTINATION ${PLUGIN_INSTALL_DIR})
 
@@ -21,7 +21,7 @@
 
 kde4_add_plugin(kurisearchfilter ${kurisearchfilter_PART_SRCS})
 
-target_link_libraries(kurisearchfilter ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS})
+target_link_libraries(kurisearchfilter ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS} ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS kurisearchfilter DESTINATION ${PLUGIN_INSTALL_DIR})
 
--- a/kurifilter-plugins/localdomain/CMakeLists.txt
+++ b/kurifilter-plugins/localdomain/CMakeLists.txt
@@ -4,7 +4,7 @@
 
 kde4_add_plugin(localdomainurifilter ${localdomainurifilter_PART_SRCS})
 
-target_link_libraries(localdomainurifilter ${KDE4_KDECORE_LIBS} ${KDE4_KIO_LIBS})
+target_link_libraries(localdomainurifilter ${KDE4_KDECORE_LIBS} ${KDE4_KIO_LIBS} ${QT_QTDBUS_LIBRARY})
 if (HAVE_NSL_LIBRARY)
 target_link_libraries(localdomainurifilter nsl)
 endif (HAVE_NSL_LIBRARY)
--- a/kurifilter-plugins/shorturi/CMakeLists.txt
+++ b/kurifilter-plugins/shorturi/CMakeLists.txt
@@ -9,7 +9,7 @@
 kde4_add_plugin(kshorturifilter ${kshorturifilter_PART_SRCS})
 
 
-target_link_libraries(kshorturifilter  ${KDE4_KIO_LIBS})
+target_link_libraries(kshorturifilter  ${KDE4_KIO_LIBS} ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS kshorturifilter  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/nepomuk/kcm/CMakeLists.txt
+++ b/nepomuk/kcm/CMakeLists.txt
@@ -23,6 +23,9 @@
 
 target_link_libraries(kcm_nepomuk
   ${STRIGI_STRIGIQTDBUSCLIENT_LIBRARY}
+  ${QT_QTDBUS_LIBRARY}
+  ${QT_QTXML_LIBRARY}
+  ${KDE4_KDECORE_LIBS}
   ${KDE4_KDEUI_LIBS}
   ${KDE4_KIO_LIBS}
   )
--- a/nepomuk/services/strigi/CMakeLists.txt
+++ b/nepomuk/services/strigi/CMakeLists.txt
@@ -17,6 +17,9 @@
 
 target_link_libraries(nepomukstrigiservice
   ${STRIGI_STRIGIQTDBUSCLIENT_LIBRARY}
+  ${QT_QTXML_LIBRARY}
+  ${QT_QTDBUS_LIBRARY}
+  ${KDE4_KDECORE_LIBS}
   ${KDE4_KDEUI_LIBS}
   ${NEPOMUK_LIBRARIES}
   )
--- a/phonon/xine/kcm/CMakeLists.txt
+++ b/phonon/xine/kcm/CMakeLists.txt
@@ -4,7 +4,7 @@
 include_directories(${XINE_INCLUDE_DIR})
 
 kde4_add_plugin(kcm_phononxine ${kcmphononxine_SRCS})
-target_link_libraries(kcm_phononxine ${KDE4_KDEUI_LIBS} ${XINE_LIBRARY})
+target_link_libraries(kcm_phononxine ${KDE4_KDEUI_LIBS} ${XINE_LIBRARY} ${QT_QTDBUS_LIBRARY})
 install(TARGETS kcm_phononxine DESTINATION ${PLUGIN_INSTALL_DIR})
 
 install(FILES kcm_phononxine.desktop DESTINATION ${SERVICES_INSTALL_DIR})
--- a/soliduiserver/CMakeLists.txt
+++ b/soliduiserver/CMakeLists.txt
@@ -13,7 +13,7 @@
 kde4_add_plugin(kded_soliduiserver ${kded_soliduiserver_SRCS})
 
 
-target_link_libraries(kded_soliduiserver ${KDE4_SOLID_LIBS}  ${KDE4_KIO_LIBS})
+target_link_libraries(kded_soliduiserver ${KDE4_SOLID_LIBS}  ${KDE4_KIO_LIBS} ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS kded_soliduiserver DESTINATION ${PLUGIN_INSTALL_DIR})
 
